src/main/java/frc/robot/commands/AutoBalance.java

```diff
@@ -52,35 +61,53 @@ public void execute() {
 
     m_drivePower = Math.min(m_balanceController.calculate(m_currentAngle), 1);
     // Limit max power
-    m_drivePower = MathUtil.clamp(m_drivePower, -0.6, 0.6);
+    m_drivePower = MathUtil.clamp(m_drivePower, -0.5, 0.5);
 
-    double rollRateBalance = m_drive.getGyroRollPitch();
+    double rollRateBalance = m_drive.getGyroPitchRate();
     // if robot is tilting forward, stop moving so we don't overshoot the balance
     if (Math.abs(rollRateBalance) >= 0.3) {
-//      m_drivePower = Math.copySign(0.0, -m_drive.getGyroPitch().getDegrees());
+      m_balanceTimer.start();
+      m_drivePower = Math.copySign(0.15, -m_drive.getGyroPitch().getDegrees());
+    }
+
+    if (m_balanceTimer.hasElapsed(0.01)) {
+//      m_drivePower = Math.copySign(0.15, -m_drive.getGyroPitch().getDegrees());
       m_drivePower = 0.0;
-//      SmartDashboard.putNumber("Stop Balancing", 1);
+    }
+    if (m_balanceTimer.hasElapsed(0.75)) {
+      m_balanceTimer.stop();
+      m_balanceTimer.reset();
```

Comment:
This shouldn't be timer based, it should be based off the robot pitch movement.

Reply:
That can be done.